### PR TITLE
Aegis 224 richw

### PR DESCRIPTION
**Pull request by richweclg**
223, 224, and 232
@TanujaG @loahou04 @marcdonovan

### CODE REVIEW

**Comment by TanujaG** [comment]
approved

**Comment by loahou04** [comment]
approved

File: src/js/NotificationApi.js [hunk 1]
```diff
@@ -11,21 +11,29 @@ function parseResponse(response) {
...
 	const archivedNotificationsList = userNotifications.filter((notification) => {
-		const result = JSON.parse(notification.payload.message);
-		notification.message = result;
-		return (notification.hasOwnProperty('notificationType') && notification.notificationType === 'inbrowser' && notification.status === 'ARCHIVED');
+		try {
+			const result = JSON.parse(notification.payload.message);
+			notification.message = result;
+			return (notification.hasOwnProperty('notificationType') && notification.notificationType === 'inbrowser' && notification.status === 'ARCHIVED');
+		} catch (e) {
+			console.log(`Error parsing payload message!\n${e}\nPayload message:\n${notification.payload.message}`);
```
Comment (marcdonovan) [inline src/js/NotificationApi.js]: some browsers do not have a console

File: src/js/NotificationList.js [hunk 2]
```diff
@@ -28,15 +28,21 @@ export default class NotificationList extends React.Component {
 	 * Render
 	 **/
 	render() {
+		const titleLength = 46;
```
Comment (marcdonovan) [inline src/js/NotificationList.js]: should we have a constants file for these?

Comment (loahou04) [inline src/js/NotificationList.js]: I actually thought about it of even having them as config...but honestly...YAGNI :P.  They aren't used in multiple places....

**Pull request merged**